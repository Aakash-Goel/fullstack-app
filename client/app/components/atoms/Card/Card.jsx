import React from 'react';
import { object, node, string } from 'prop-types';
import classnames from 'classnames';

import withStyles from '@material-ui/core/styles/withStyles';
import { Card as MuiCard } from '@material-ui/core';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  classes: object.isRequired,
  children: node.isRequired,
  className: string,
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  className: '',
};

const style = {
  cardRoot: {
    padding: '8px',
    borderTop: '4px solid green', // needs to change, it should be handle by variations
  },
};

const Card = ({ ...props }) => {
  const { classes, children, className, ...rest } = props;

  const cardClasses = classnames({
    [classes.cardRoot]: true,
    [className]: className,
  });

  return (
    <MuiCard {...rest} className={cardClasses}>
      {children}
    </MuiCard>
  );
};

Card.propTypes = propTypes;
Card.defaultProps = defaultProps;

export default withStyles(style)(Card);
