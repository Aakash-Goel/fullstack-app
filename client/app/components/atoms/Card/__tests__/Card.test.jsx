import React from 'react';
import TestRenderer from 'react-test-renderer';

import Card from '../Card';

describe('Card atom', () => {
  it('should render correctly', () => {
    const cardWrapper = TestRenderer.create(
      <Card>Card content goes here</Card>
    ).toJSON();
    expect(cardWrapper).toMatchSnapshot();
  });
});
