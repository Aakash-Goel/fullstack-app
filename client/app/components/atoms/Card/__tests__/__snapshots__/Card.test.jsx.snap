// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card atom should render correctly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation1-6 MuiPaper-rounded-4 MuiCard-root-2 Card-cardRoot-1"
>
  Card content goes here
</div>
`;
